xwiki-batchimport-api/src/main/java/org/xwiki/batchimport/internal/DefaultBatchImport.java

```diff
@@ -1348,7 +1348,7 @@ public Document marshalDataToDocumentObjects(DocumentReference pageName, Map<Str
                 if (!key.startsWith("doc.")) {
                     PropertyInterface prop = defaultClass.get(key);
 
-                    if (value != null && (!StringUtils.isEmpty(stringValue) || overwrite == Overwrite.UPDATE_EVEN_WHEN_EMPTY_INPUT_CELL)) {
+                    if (!StringUtils.isEmpty(stringValue) || config.getHonorEmptyValues()) {
                         boolean addtotags = false;
```

Comment:
The TODO from line 1346 above needs to be removed / updated to reflect the new behaviour.

Reply:
Right, I removed it in the latest commit.